be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -70,62 +72,68 @@ sealed class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
-        val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
-        val unknownAnswer: String = "Database actor returned an unknown answer"
-
-        //Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
-        def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
-          attendees match {
-            case Nil => Left(rpcMessage)
-            case head::tail =>
-              //a closeRollCall is always sent in the lao's main channel so we are allowed to do this
-              val socialChannel: String = generateSocialChannel(rpcMessage.getParamsChannel, head)
-              val ask: Future[GraphMessage] = (dbActor ? DbActor.CreateChannel(Channel(socialChannel), ObjectType.CHIRP)).map {
-                case DbActorAck() => createAttendeeChannels(tail, rpcMessage)
-                //the distinction between the NAck cases this is to prevent errors for now within a LAO for successive Roll Calls with the same participants since the public key should not change, as the channel already exists
-                case DbActorNAck(code, description) if code == ErrorCodes.INVALID_RESOURCE.id => createAttendeeChannels(tail, rpcMessage)
-                case DbActorNAck(code, description) => Right(PipelineError(code, description, rpcMessage.id))
-                case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, unknownAnswer, rpcMessage.id))
+    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
+    Await.result(ask, duration) match {
+      case Left(msg) => {
+        rpcMessage.getParamsMessage match {
+          case Some(message: Message) =>
+            val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
+
+            //Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
+            def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
+              attendees match {
+                case Nil => Left(rpcMessage)
+                case head::tail =>
+                  //a closeRollCall is always sent in the lao's main channel so we are allowed to do this
+                  val socialChannel: String = generateSocialChannel(rpcMessage.getParamsChannel, head)
+                  val ask: Future[GraphMessage] = (dbActor ? DbActor.CreateChannel(Channel(socialChannel), ObjectType.CHIRP)).map {
+                    case DbActorAck() => createAttendeeChannels(tail, rpcMessage)
+                    //the distinction between the NAck cases this is to prevent errors for now within a LAO for successive Roll Calls with the same participants since the public key should not change, as the channel already exists
+                    case DbActorNAck(code, description) if code == ErrorCodes.INVALID_RESOURCE.id => createAttendeeChannels(tail, rpcMessage)
```

Comment:
The attendee's pop token should change at every roll call, so I'm it sure you need this 🤔

Reply:
Yeah, this was originally left here in the case the front-end was using the PublicKeys and not the tokens yet (before the in-meeting discussion) but now that it was cleared it can be removed